syntax = "proto3";

package test;

message TestMsg {
  string from = 1;
  repeated string to = 2;
  oneof test_oneof {
    string message_text = 3;
    // Other types of payload could be added here, e.g. images
  }
}

message HeartbeatMsg {
  string from = 1;
  repeated string active_users = 2;
}